pkg/register/registerDelegationProof_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) IOTIC LABS LIMITED. All rights reserved. Licensed under the Apache License, Version 2.0.
+
+package register_test
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/proof"
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/register"
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/test"
+)
+
+func Test_validate_delegation_fails_if_invalid_controller_issuer(t *testing.T) {
+	invalidRegisteredProof := &register.RegisterDelegationProof{
+		Controller: "notAnIssuer",
+	}
+	err := register.ValidateDelegation(nil, "registeredDid", invalidRegisteredProof)
+	assert.ErrorContains(t, err, "invalid issuer string")
+}
+
+func Test_validate_delegation_fails_if_can_not_get_controller_doc(t *testing.T) {
+	invalidRegisteredProof := &register.RegisterDelegationProof{
+		Controller: "did:iotics:iotTdY27bo6ycsVs49T35QuPHZfjrJ3P41NR#user-name",
+	}
+	resolver := test.NewInMemoryResolver()
+	err := register.ValidateDelegation(resolver, "registeredDid", invalidRegisteredProof)
+	assert.ErrorContains(t, err, "document not found")
+}
+
+func Test_validate_delegation_fails_controller_public_key_not_found(t *testing.T) {
+	controllerDoc, _, _ := test.HelperGetRegisterDocument()
+	registeredProof := &register.RegisterDelegationProof{
+		Controller: "did:iotics:iotTdY27bo6ycsVs49T35QuPHZfjrJ3P41NR#different-issuer",
+	}
+	resolver := test.NewInMemoryResolver(controllerDoc)
+	err := register.ValidateDelegation(resolver, "registeredDid", registeredProof)
+	assert.ErrorContains(t, err, "controller public key #different-issuer not found")
+}
+
+func Test_validate_delegation_fails_if_content_does_not_match_registered_id_or_controller_id(t *testing.T) {
+	controllerDoc, issuer, keyPair := test.HelperGetRegisterDocument()
+	cryptoProof, err := proof.NewProof(keyPair.PrivateKey, issuer.Did, issuer.Name, []byte("content no match"))
+	assert.NilError(t, err)
+	registeredProof, err := register.NewRegisterDelegationProof("#ProofName", issuer.String(), cryptoProof.Signature, false)
+	assert.NilError(t, err)
+	resolver := test.NewInMemoryResolver(controllerDoc)
+	err = register.ValidateDelegation(resolver, "did:aregisteredDID", registeredProof)
+	assert.ErrorContains(t, err, "invalid signature")
+}
+
+func Test_validate_delegation_succeeds_if_registered_proof_content_matches_registered_id(t *testing.T) {
+	controllerDoc, issuer, keyPair := test.HelperGetRegisterDocument()
+	registeredID := "did:aregisteredDID"
```

Comment:
Just to make sure this differs from the tests below you might want to update this DID string so it isn't the same as the controllerDoc.ID

Reply:
https://github.com/Iotic-Labs/iotics-identity-go/pull/42/commits/496cf6359da16a2e3e8931e9724305d4cedbf418